backend/src/agent/persistence/main.go

```diff
@@ -119,14 +126,35 @@ func main() {
 		log.Fatalf("Error starting Service Account Token Refresh Ticker due to: %v", err)
 	}
 
+	var tlsCfg *tls.Config
+	if mlPipelineServiceTLSEnabled {
+		caCertPool, err := x509.SystemCertPool()
```

Comment:
This code block seems to be the same in 4 places. Could you extract this out to a helper function please?